gordo_components/cli/cli.py

```diff
@@ -238,26 +248,32 @@ def build(
                 )
 
                 model_location = provide_saved_model(
-                    name,
-                    model_config,
-                    data_config,
-                    metadata,
-                    output_dir,
-                    model_register_dir,
+                    output_dir=output_dir,
+                    model_register_dir=model_register_dir,
                     replace_cache=True,
-                    evaluation_config=evaluation_config,
+                    **build_kwargs,
                 )
                 saved_metadata = load_metadata(model_location)
                 all_scores = get_all_score_strings(saved_metadata)
 
             for score in all_scores:
                 print(score)
+
+        # If enabled, configure remote logging to AzureML, otherwise logs locally
+        workspace_kwargs = get_workspace_kwargs() if enable_remote_logging else {}
+        service_principal_kwargs = get_spauth_kwargs() if enable_remote_logging else {}
+        mlflow_logger = get_mlflow_logger(
+            name, model_key, workspace_kwargs, service_principal_kwargs
+        )
+
+        mlflow_logger(metadata)
```

Comment:
This confuses me, is it a "factory" that produces a function? And then call the function?

Reply:
I am just a bit stuck in the non-functional world, just ignore me - this is just fine for me 👍